### ISSUE TITLE
Actionlint does not alert on a mis-terminated expression for `if`

### ISSUE THREAD

**Issue by razzmatazz** [rhysd/actionlint#373]
For example, I have this code in the workflow file:

```yaml
(...)
jobs:
  jobX:
    (...)

  jobY:
>>> if: ${{ needs.jobA.outputs.A == 'true' }}
>>>         || needs.jobB.outputs.B == 'true' }}
    needs:
      - jobX
    steps:
(...)
```

actionlint does not report it that there is a second line after `if` that somehow makes github actions ignore the condition under `if:` and will run this job unconditionally
